client/eth.go

```diff
@@ -18,52 +18,41 @@ var (
 // EthClient provides access to Coreth API
 type EthClient struct {
 	ethclient.Client
-	rpc         rpc.Requester
+	rpc         *rpc.Client
 	traceConfig *tracers.TraceConfig
 }
 
 // NewEthClient returns a new EVM client
-func NewEthClient(endpoint string) (*EthClient, error) {
+func NewEthClient(ctx context.Context, endpoint string) (*EthClient, error) {
 	endpointURL := fmt.Sprintf("%s%s", endpoint, prefixEth)
 
-	c, err := ethclient.Dial(endpointURL)
-	if err != nil {
-		return nil, err
-	}
+	c, err := rpc.DialContext(ctx, endpointURL)
 
 	return &EthClient{
-		Client: c,
-		rpc:    rpc.NewRPCRequester(endpoint),
+		Client: ethclient.NewClient(c),
```

Comment:
Could panic if `c` is nil...just return err

Reply:
This wouldn't panic - reverting these changes so it's more obvious